mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +327,50 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after training.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.published_keys is None:
+            return
+
+        if self.save_last:
+            last_ckpt = runner.message_hub.get_info('last_ckpt')
+            assert last_ckpt ('Did not find last_checkpoint to be resumed.')
+            self._publish_model(runner, last_ckpt)
+
+        if self.save_best is not None:
+            if not self.best_ckpt_path:
+                raise RuntimeError(xxx)
+            self._publish_model(runner, best_ckpt)
+
+    def _publish_model(self, runner, out_file: str) -> None:
+
+        if published_keys is None:
+            return
+        checkpoint = runner.load_checkpoint(out_file)
```

Comment:
`runner.load_checkpoint` will load the checkpoint to model which is not expected.
we can use `_load_checkpoint` with `from mmengine.runner.checkpoint import _load_checkpoint`